### DISCUSSION TITLE
Wagtail Custom Revisions Diffing

### DISCUSSION THREAD

**Discussion by skelzor06** [wagtail/wagtail#11730] [Development]
Hello,

I've been using Wagtail for while now and I'm trying to change the behavior of revision diffing. No it turns the RTF into a flat string and compare, right? Would be possible for me, somehow, to maybe override RichTextFieldComparison in my own project to convert html to a markdown and then compare? Since markdown is also a kind of text, maybe it will show a better comparison.

**Comment by SebCorbin** [comment]
You can try this (`markdown_from_html()` should be your function converting HTML to Markdown):

```python
# in wagtail_hooks.py
from wagtail.admin.compare import register_comparison_class, TextFieldComparison

class MarkdownComparison(TextFieldComparison):
    def htmldiff(self):
        return diff_text(
            markdown_from_html(self.val_a), markdown_from_html(self.val_b)
        ).to_html()


register_comparison_class(RichTextField, comparison_class=MarkdownComparison)
```

It will apply to all RTF fields, so if you need more control, you may have to subclass RTF and pass `exact_class=True` to `register_comparison_class`

**Reply by skelzor06** [reply]
@SebCorbin When I edit the page and them compare with a different version, its possible to know if Wagtail is actually using this comparison?

**Reply by SebCorbin** [reply]
You may add a breakpoint or something like that in this code, but if you need more details, you'll need to dig into the source:
 - [the revision view](https://github.com/wagtail/wagtail/blob/007d5b048d061266c696c5b72c6c4530382351a0/wagtail/admin/views/generic/models.py#L1141)
 - [the comparison made in FieldPanel](https://github.com/wagtail/wagtail/blob/007d5b048d061266c696c5b72c6c4530382351a0/wagtail/admin/panels/field_panel.py#L352)
 - [how the panel searches for comparison_class](https://github.com/wagtail/wagtail/blob/007d5b048d061266c696c5b72c6c4530382351a0/wagtail/admin/panels/field_panel.py#L63)
 - [where the registry of comparison classes is defined](https://github.com/wagtail/wagtail/blob/007d5b048d061266c696c5b72c6c4530382351a0/wagtail/admin/compare.py#L17)

**Reply by laymonage** [reply]
or perhaps add some temporary test string in the rendered HTML and see that it shows up in the view...?
